alacritty_terminal/src/term/mod.rs

```diff
@@ -810,15 +790,14 @@ impl SizeInfo {
         Column(((self.width - 2. * self.padding_x) / self.cell_width) as usize)
     }
 
-    pub fn contains_point(&self, x: usize, y: usize, include_padding: bool) -> bool {
-        if include_padding {
-            x < self.width as usize && y < self.height as usize
-        } else {
-            x < (self.width - self.padding_x) as usize
-                && x >= self.padding_x as usize
-                && y < (self.height - self.padding_y) as usize
-                && y >= self.padding_y as usize
-        }
+    /// Check if coordinates are inside the terminal grid.
+    ///
+    /// The padding is not counted as part of the grid.
+    pub fn contains_point(&self, x: usize, y: usize) -> bool {
+        x < (self.width - self.padding_x) as usize
+            && x >= self.padding_x as usize
```

Comment:
What the idea behind half opened ranges here? I think we should go with `(,)` instead of `[,)`, right?

Reply:
This is not code I've written myself, but it's correct the way it is from what I can tell.

Just to 100% confirm it, I made this little 'sketch' which should show it pretty clearly:
```
X X 2 3 4 5 6 7 X X

width: 10

padding: 2

x < 8 -> x < (10 - 2) -> x < (width - padding)

x >= 2 -> x >= padding
```

I think that's a pretty simple way to show what's going on here, should hopefully make it clear.

Reply:
Ah, make sense.